cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -160,18 +160,30 @@ Status ResolveOneFieldRef(
     const std::unordered_map<std::string, const SchemaField*>& field_lookup,
     const std::unordered_set<std::string>& duplicate_fields,
     std::vector<int>* columns_selection) {
-  if (const std::string* name = field_ref.name()) {
+  auto resolve_field_ref = [&](const std::string* name) -> Status {
     auto it = field_lookup.find(*name);
     if (it != field_lookup.end()) {
       AddColumnIndices(*it->second, columns_selection);
     } else if (duplicate_fields.find(*name) != duplicate_fields.end()) {
-      // We shouldn't generally get here because SetProjection will reject such references
+      // We shouldn't generally get here because SetProjection will reject such
+      // references
       return Status::Invalid("Ambiguous reference to column '", *name,
                              "' which occurs more than once");
     }
+    return Status::OK();
+  };
+
+  if (const std::string* name = field_ref.name()) {
+    RETURN_NOT_OK(resolve_field_ref(name));
     // "Virtual" column: field is not in file but is in the ScanOptions.
     // Ignore it here, as projection will pad the batch with a null column.
     return Status::OK();
+  } else if (const FieldPath* path = field_ref.field_path()) {
+    int index = path->indices()[0];
```

Comment:
Should we explicitly reject nested field paths here for now?

Reply:
That’s an unclear point to me yet. That’s why I also describe the current work as an  initial fix. How are we using nested fields so far? I was looking into it a bit.

Reply:
I mean, it's not fully implemented, so we should at least error or include a test case or both.

What do you mean by using? It's supported, but not very well, there are some prerequisite JIRAs needed to get things fully working but those have languished.

Reply:
I got your point. Let me error and add a few tests.